Decision Id in check response

PietervdWerk opened the pull request:
Fixes https://github.com/open-policy-agent/opa/issues/6519 by adding decision id to the dynamic metadata.

ashutosh-narkar commented on internal/internal.go:
```diff
@@ -557,6 +557,18 @@ func (p *envoyExtAuthzGrpcServer) check(ctx context.Context, req interface{}) (*
+			StringValue: result.DecisionID,
+		},
+	}
+
```
What if we first check if the decision id is defined and if it is then we include it in the dynamic metadata? Currently we'll include dynamic metadata always and it could possibly not be that useful in some cases.

PietervdWerk replied:
The decision id is always a uuid4, if it errors the error is returned by the check function so it looks to me like that would be a redundant check.

ashutosh-narkar reviewed (commented):
Hey @PietervdWerk thanks for picking this up. If you could add some unit tests that would be great.

ashutosh-narkar commented:
@PietervdWerk the changes look good. Can you please sign-off your commit and we can get this in. Thanks.

srenatus reviewed (approved):
Thanks!! 👏

ashutosh-narkar commented:
@PietervdWerk something may gotten missed up when you rebased. The `golang.org/x/tools` update should not be part of your change. Can you please fix that and push a single commit with your change. Also thanks for fixing the sign-off.

PietervdWerk commented:
> @PietervdWerk something may gotten missed up when you rebased. The `golang.org/x/tools` update should not be part of your change. Can you please fix that and push a single commit with your change. Also thanks for fixing the sign-off.

I've cleaned the main history and rebased my commit on it. Sorry about that, first time signing off on a commit. Should be all good now :).

ashutosh-narkar reviewed (approved):
LGTM. Thanks @PietervdWerk!
